_test/test_sqw_pageOpMethods/test_split.m

```diff
@@ -0,0 +1,157 @@
+classdef test_split< TestCase
+    %
+    % Validate sqw object replication
+    %
+
+    properties
+        this_dir;
+        sqw_source = 'common_data/sqw_4d.sqw'
+
+        source_sqw4D;
+    end
+
+    methods
+        function obj=test_split(name)
+            if ~exist('name','var')
+                name = 'test_split';
+            end
+            obj=obj@TestCase(name);
+            obj.this_dir = fileparts(mfilename('fullpath'));
+            [fp,fn,fe] = fileparts(obj.sqw_source);
+            source_data = fullfile(fileparts(obj.this_dir),fp,[fn,fe]);
+            obj.sqw_source   = source_data;
+            obj.source_sqw4D = read_sqw(source_data);
+
+        end
+
+        function test_split_pix_filebacked_permanent_res(obj)
+            n_pix = obj.source_sqw4D.npixels;
+
+            targ_folder = fullfile(tmp_dir,'split_fb_targ');
+            clFiles = onCleanup(@()rmdir(targ_folder,'s'));
+            w_spl = split(obj.source_sqw4D,'-filebacked',targ_folder);
+
+            assertEqual(numel(w_spl),23);
+
+            n_split_pix = 0;
+            for i=1:numel(w_spl)
+                assertTrue(w_spl(i).is_filebacked);
+                keys = w_spl(i).runid_map.keys;
+                assertEqual(numel(keys),1);
+                id = unique(w_spl(i).pix.run_idx);
+                assertEqual(keys{1},id);
+                assertEqual(w_spl(i).experiment_info.expdata.run_id,id);
+                n_split_pix  = n_split_pix +w_spl(i).npixels;
+            end
+            assertEqual(n_pix,n_split_pix);
+
+            assertTrue(isfolder(targ_folder));
+
+            % check that filebacked objects are temporary objects in this
+            % case
+            files = cell(1,numel(w_spl));
+            for i=1:numel(files)
+                files{i} = w_spl(i).full_filename;
+
+                assertTrue(is_file(files{i}));
+                [~,~,fe] = fileparts(files{i});
+                assertEqual(fe,'.sqw');
```

Comment:
Do we care that the file extension is `.sqw`?

Reply:
yes, this indicates that it is not temporary file but remains permantent. Relatively complex logic worh unit test

Reply:
If you're checking that it's *not* a temporary file, why does the comment say otherwise? Extra comments would be helpful.